sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -227,6 +227,7 @@ abstract class BinaryComparison extends BinaryOperator with Predicate {
 
   override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
     if (ctx.isPrimitiveType(left.dataType)
+        && left.dataType != BooleanType // java boolean doesn't support > or < operator
```

Comment:
yes, but without test against `BooleanType`, it would use 
```scala
// faster version
defineCodeGen(ctx, ev, (c1, c2) => s"$c1 $symbol $c2")
```
rather than the else part, therefore, `genComp` is not used.